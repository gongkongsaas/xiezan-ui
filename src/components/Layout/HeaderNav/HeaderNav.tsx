import { defineComponent, PropType } from 'vue'
import { UserInfoType } from '@/types/userInfoType'

export default defineComponent({
  props: {
    companyName: {
      type: String,
      default: '运营中心菜单',
    },
    userInfoData: {
      type: Object as PropType<UserInfoType>,
      default: {},
    },
  },
  setup(props) {
    return () => {
      return (
        <div class="header-nav">
          <div class="header-information">
            <a class="logo" href="/">
              <img src="/upload/logo/favorites_icon.png" alt="LOGO" />
            </a>
          </div>
          <div class="header-option">
            <div class="mall-company-name">
              <div class="company-name">{props.companyName}</div>
            </div>
            <div class="search-box"></div>
            <ul class="option-box">
              <li class="home">
                <a href="/">商城</a>
              </li>
              <li class="task">
                <a
                  href="/approval/approval-center/approval-unfinished-management"
                  target="_blank"
                >
                  待办
                </a>
              </li>
              <li class="message">
                <a href="/message/message/message-list" target="_blank">
                  消息
                </a>
              </li>
              <li class="service">
                <a
                  href="http://q.url.cn/ABRGOl?_type=wpa&qidian=true"
                  target="_blank"
                >
                  客服
                </a>
              </li>
              <li class="help">
                <a href="/help/help/index" target="_blank">
                  帮助
                </a>
              </li>
              <li class="info">
                <a href="#">
                  <i class="iconfont">&#xe976;</i>
                  <span>{props.userInfoData.userName}</span>
                </a>
              </li>
            </ul>
          </div>
        </div>
      )
    }
  },
})
