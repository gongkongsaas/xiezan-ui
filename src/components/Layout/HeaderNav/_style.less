.header-nav {
  display: flex;
  width: 100%;
  height: 50px;
  font-size: 14px;
  color: #333;
  background: #fff;

  .operation-main-information {
    width: 90px;
    background-color: #1e2834;
  }

  .mall-company-name {
    display: flex;
    flex: 1;
    border-bottom: 1px #e6e8ec solid;
    line-height: 50px;
    cursor: pointer;

    .company-name {
      padding: 0 20px;
    }
  }

  .option-box {
    width: auto;
    padding-left: 50px;
    border-bottom: 1px #e6e8ec solid;
    line-height: 50px;
  }
}
