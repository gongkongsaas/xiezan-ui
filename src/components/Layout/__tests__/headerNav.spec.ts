import { shallowMount } from '@vue/test-utils'
import HeaderNav from '../header-nav/headerNav'

describe('HeaderNav', () => {
  it('mounted health', () => {
    const wrapper = shallowMount(HeaderNav, {
      props: {
        companyName: '测试菜单名',
        userInfoData: {
          accountType: 1,
          avatarFilePath: '/path',
          companyName: 'testCompany',
          masterSign: false,
          operatorSign: false,
          sellerSign: true,
          userName: '测试名字',
        },
      },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
