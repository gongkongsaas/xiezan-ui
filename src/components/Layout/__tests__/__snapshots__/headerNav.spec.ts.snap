// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderNav mounted health 1`] = `
<div class="header-nav">
  <div class="header-information"><a class="logo" href="/"><img src="/upload/logo/favorites_icon.png" alt="LOGO"></a></div>
  <div class="header-option">
    <div class="mall-company-name">
      <div class="company-name">测试菜单名</div>
    </div>
    <div class="search-box"></div>
    <ul class="option-box">
      <li class="home"><a href="/">商城</a></li>
      <li class="task"><a href="/approval/approval-center/approval-unfinished-management" target="_blank">待办</a></li>
      <li class="message"><a href="/message/message/message-list" target="_blank">消息</a></li>
      <li class="service"><a href="http://q.url.cn/ABRGOl?_type=wpa&amp;qidian=true" target="_blank">客服</a></li>
      <li class="help"><a href="/help/help/index" target="_blank">帮助</a></li>
      <li class="info"><a href="#"><i class="iconfont"></i><span>测试名字</span></a></li>
    </ul>
  </div>
</div>
`;
